In [1]:
#import library
import pandas as pd

In [2]:
#Read the dataset
dataset=pd.read_csv("Placement.csv")

In [3]:
dataset

sl_no gender  ssc_p    ssc_b  hsc_p    hsc_b     hsc_s  degree_p  \
0        1      M  67.00   Others  91.00   Others  Commerce     58.00   
1        2      M  79.33  Central  78.33   Others   Science     77.48   
2        3      M  65.00  Central  68.00  Central      Arts     64.00   
3        4      M  56.00  Central  52.00  Central   Science     52.00   
4        5      M  85.80  Central  73.60  Central  Commerce     73.30   
..     ...    ...    ...      ...    ...      ...       ...       ...   
210    211      M  80.60   Others  82.00   Others  Commerce     77.60   
211    212      M  58.00   Others  60.00   Others   Science     72.00   
212    213      M  67.00   Others  67.00   Others  Commerce     73.00   
213    214      F  74.00   Others  66.00   Others  Commerce     58.00   
214    215      M  62.00  Central  58.00   Others   Science     53.00   

      degree_t workex  etest_p specialisation  mba_p      status    salary  
0     Sci&Tech     No     55.0         Mkt&HR  58.80      Placed  270000.0  
1     Sci&Tech    Yes     86.5        Mkt&Fin  66.28      Placed  200000.0  
2    Comm&Mgmt     No     75.0        Mkt&Fin  57.80      Placed  250000.0  
3     Sci&Tech     No     66.0         Mkt&HR  59.43  Not Placed       NaN  
4    Comm&Mgmt     No     96.8        Mkt&Fin  55.50      Placed  425000.0  
..         ...    ...      ...            ...    ...         ...       ...  
210  Comm&Mgmt     No     91.0        Mkt&Fin  74.49      Placed  400000.0  
211   Sci&Tech     No     74.0        Mkt&Fin  53.62      Placed  275000.0  
212  Comm&Mgmt    Yes     59.0        Mkt&Fin  69.72      Placed  295000.0  
213  Comm&Mgmt     No     70.0         Mkt&HR  60.23      Placed  204000.0  
214  Comm&Mgmt     No     89.0         Mkt&HR  60.22  Not Placed       NaN  

[215 rows x 15 columns]

In [4]:
dataset.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 215 entries, 0 to 214
Data columns (total 15 columns):
sl_no             215 non-null int64
gender            215 non-null object
ssc_p             215 non-null float64
ssc_b             215 non-null object
hsc_p             215 non-null float64
hsc_b             215 non-null object
hsc_s             215 non-null object
degree_p          215 non-null float64
degree_t          215 non-null object
workex            215 non-null object
etest_p           215 non-null float64
specialisation    215 non-null object
mba_p             215 non-null float64
status            215 non-null object
salary            148 non-null float64
dtypes: float64(6), int64(1), object(8)
memory usage: 25.3+ KB


In [5]:
#Quan=["ssc_p","hsc_p","etest_p","degree_p","mba_p","salary"]
#Qual=["gender","ssc_b","hsc_b","degree_t"]

In [6]:
dataset.columns

Index(['sl_no', 'gender', 'ssc_p', 'ssc_b', 'hsc_p', 'hsc_b', 'hsc_s',
       'degree_p', 'degree_t', 'workex', 'etest_p', 'specialisation', 'mba_p',
       'status', 'salary'],
      dtype='object')

In [7]:
#find dataset type
dir(dataset)

['T',
 '_AXIS_ALIASES',
 '_AXIS_IALIASES',
 '_AXIS_LEN',
 '_AXIS_NAMES',
 '_AXIS_NUMBERS',
 '_AXIS_ORDERS',
 '_AXIS_REVERSED',
 '__abs__',
 '__add__',
 '__and__',
 '__array__',
 '__array_priority__',
 '__array_wrap__',
 '__bool__',
 '__class__',
 '__contains__',
 '__copy__',
 '__deepcopy__',
 '__delattr__',
 '__delitem__',
 '__dict__',
 '__dir__',
 '__div__',
 '__doc__',
 '__eq__',
 '__finalize__',
 '__floordiv__',
 '__format__',
 '__ge__',
 '__getattr__',
 '__getattribute__',
 '__getitem__',
 '__getstate__',
 '__gt__',
 '__hash__',
 '__iadd__',
 '__iand__',
 '__ifloordiv__',
 '__imod__',
 '__imul__',
 '__init__',
 '__init_subclass__',
 '__invert__',
 '__ior__',
 '__ipow__',
 '__isub__',
 '__iter__',
 '__itruediv__',
 '__ixor__',
 '__le__',
 '__len__',
 '__lt__',
 '__matmul__',
 '__mod__',
 '__module__',
 '__mul__',
 '__ne__',
 '__neg__',
 '__new__',
 '__nonzero__',
 '__or__',
 '__pos__',
 '__pow__',
 '__radd__',
 '__rand__',
 '__rdiv__',
 '__reduce__',
 '__reduce_ex__',
 '__repr__',
 

In [8]:
dataset["ssc_p"].dtypes

dtype('float64')

In [9]:
#create the for loop 
for columnName in dataset.columns:
    print(columnName)

sl_no
gender
ssc_p
ssc_b
hsc_p
hsc_b
hsc_s
degree_p
degree_t
workex
etest_p
specialisation
mba_p
status
salary


In [10]:
#Create the ifelse condition
if(dataset["ssc_p"].dtype=='O'):
    print("qual")
else:
    print("quan")

quan


In [11]:
quan=[]
qual=[]
for columnName in dataset.columns:
    #print(columnName)
    if(dataset[columnName].dtype=='O'):
        #print("qual")
        qual.append(columnName)
    else:
        #print("quan")
        quan.append(columnName)
        


In [12]:
quan

['sl_no', 'ssc_p', 'hsc_p', 'degree_p', 'etest_p', 'mba_p', 'salary']

In [13]:
qual

['gender',
 'ssc_b',
 'hsc_b',
 'hsc_s',
 'degree_t',
 'workex',
 'specialisation',
 'status']

In [14]:
#Create the function
def quanQual(dataset):
    quan=[]
    qual=[]
    for columnName in dataset.columns:
        #print(columnName)
        if(dataset[columnName].dtype=='O'):
            #print("qual")
            qual.append(columnName)
        else:
            #print("quan")
            quan.append(columnName)
    return quan,qual


    

In [15]:
quanQual(dataset)

(['sl_no', 'ssc_p', 'hsc_p', 'degree_p', 'etest_p', 'mba_p', 'salary'],
 ['gender',
  'ssc_b',
  'hsc_b',
  'hsc_s',
  'degree_t',
  'workex',
  'specialisation',
  'status'])

In [16]:
quan,qual=quanQual(dataset)

In [17]:
quan

['sl_no', 'ssc_p', 'hsc_p', 'degree_p', 'etest_p', 'mba_p', 'salary']

In [18]:
qual

['gender',
 'ssc_b',
 'hsc_b',
 'hsc_s',
 'degree_t',
 'workex',
 'specialisation',
 'status']

In [19]:
dataset

sl_no gender  ssc_p    ssc_b  hsc_p    hsc_b     hsc_s  degree_p  \
0        1      M  67.00   Others  91.00   Others  Commerce     58.00   
1        2      M  79.33  Central  78.33   Others   Science     77.48   
2        3      M  65.00  Central  68.00  Central      Arts     64.00   
3        4      M  56.00  Central  52.00  Central   Science     52.00   
4        5      M  85.80  Central  73.60  Central  Commerce     73.30   
..     ...    ...    ...      ...    ...      ...       ...       ...   
210    211      M  80.60   Others  82.00   Others  Commerce     77.60   
211    212      M  58.00   Others  60.00   Others   Science     72.00   
212    213      M  67.00   Others  67.00   Others  Commerce     73.00   
213    214      F  74.00   Others  66.00   Others  Commerce     58.00   
214    215      M  62.00  Central  58.00   Others   Science     53.00   

      degree_t workex  etest_p specialisation  mba_p      status    salary  
0     Sci&Tech     No     55.0         Mkt&HR  58.80      Placed  270000.0  
1     Sci&Tech    Yes     86.5        Mkt&Fin  66.28      Placed  200000.0  
2    Comm&Mgmt     No     75.0        Mkt&Fin  57.80      Placed  250000.0  
3     Sci&Tech     No     66.0         Mkt&HR  59.43  Not Placed       NaN  
4    Comm&Mgmt     No     96.8        Mkt&Fin  55.50      Placed  425000.0  
..         ...    ...      ...            ...    ...         ...       ...  
210  Comm&Mgmt     No     91.0        Mkt&Fin  74.49      Placed  400000.0  
211   Sci&Tech     No     74.0        Mkt&Fin  53.62      Placed  275000.0  
212  Comm&Mgmt    Yes     59.0        Mkt&Fin  69.72      Placed  295000.0  
213  Comm&Mgmt     No     70.0         Mkt&HR  60.23      Placed  204000.0  
214  Comm&Mgmt     No     89.0         Mkt&HR  60.22  Not Placed       NaN  

[215 rows x 15 columns]

# To Find Central tendency - Mean, Median, Mode

In [20]:
# Average value of currect data columns
dataset["ssc_p"].mean()

67.30339534883721

In [21]:
# Centre value of currect data columns
dataset["ssc_p"].median()

67.0

In [22]:
# Common value of currect data columns
dataset["ssc_p"].mode()[0]

62.0

In [23]:
#Create the data Table
dataset.describe()

sl_no       ssc_p       hsc_p    degree_p     etest_p       mba_p  \
count  215.000000  215.000000  215.000000  215.000000  215.000000  215.000000   
mean   108.000000   67.303395   66.333163   66.370186   72.100558   62.278186   
std     62.209324   10.827205   10.897509    7.358743   13.275956    5.833385   
min      1.000000   40.890000   37.000000   50.000000   50.000000   51.210000   
25%     54.500000   60.600000   60.900000   61.000000   60.000000   57.945000   
50%    108.000000   67.000000   65.000000   66.000000   71.000000   62.000000   
75%    161.500000   75.700000   73.000000   72.000000   83.500000   66.255000   
max    215.000000   89.400000   97.700000   91.000000   98.000000   77.890000   

              salary  
count     148.000000  
mean   288655.405405  
std     93457.452420  
min    200000.000000  
25%    240000.000000  
50%    265000.000000  
75%    300000.000000  
max    940000.000000

In [67]:
#create the DatafRame without Index
descriptive=pd.DataFrame()

In [68]:
#create the DatafRame with Index
descriptive=pd.DataFrame(index=["Mean","Median","Mode"],columns=quan)

In [71]:
descriptive

sl_no ssc_p hsc_p degree_p etest_p mba_p salary
Mean     NaN   NaN   NaN      NaN     NaN   NaN    NaN
Median   NaN   NaN     0      NaN     NaN   NaN    NaN
Mode     NaN   NaN   NaN      NaN     NaN   NaN    NaN

In [70]:
#access the selected dataset values
descriptive["hsc_p"]["Median"]=0

In [72]:
#Create the loop and find the Mean Value in All datset Columns
descriptive=pd.DataFrame(index=["Mean","Median","Mode"],columns=quan)
for columnName in quan:
    descriptive[columnName]["Mean"]=dataset[columnName].mean()

In [29]:
descriptive

sl_no    ssc_p    hsc_p degree_p  etest_p    mba_p  salary
Mean     108  67.3034  66.3332  66.3702  72.1006  62.2782  288655
Median   NaN      NaN      NaN      NaN      NaN      NaN     NaN
Mode     NaN      NaN      NaN      NaN      NaN      NaN     NaN

In [30]:
#Create the loop and find the Mean, Median Value in All datset Columns
descriptive=pd.DataFrame(index=["Mean","Median","Mode"],columns=quan)
for columnName in quan:
    descriptive[columnName]["Mean"]=dataset[columnName].mean()
    descriptive[columnName]["Median"]=dataset[columnName].median()

In [31]:
descriptive

sl_no    ssc_p    hsc_p degree_p  etest_p    mba_p  salary
Mean     108  67.3034  66.3332  66.3702  72.1006  62.2782  288655
Median   108       67       65       66       71       62  265000
Mode     NaN      NaN      NaN      NaN      NaN      NaN     NaN

In [32]:
#Create the loop and find the Mean, Median,Mode  Value in All datset Columns
descriptive=pd.DataFrame(index=["Mean","Median","Mode"],columns=quan)
for columnName in quan:
    descriptive[columnName]["Mean"]=dataset[columnName].mean()
    descriptive[columnName]["Median"]=dataset[columnName].median()
    descriptive[columnName]["Mode"]=dataset[columnName].mode()[0]
    

In [33]:
#Why find descriptive Mean, Median,Mode  Value ? Because need to find a solution as a information from the Datasets
descriptive

sl_no    ssc_p    hsc_p degree_p  etest_p    mba_p  salary
Mean     108  67.3034  66.3332  66.3702  72.1006  62.2782  288655
Median   108       67       65       66       71       62  265000
Mode       1       62       63       65       60     56.7  300000


 📊 **Placement Data Analysis – Student Performance and Salary Insights**

The dataset represents the academic and placement performance of 108 students, focusing on their percentage scores and salaries after placement. Here’s a summary based on the central tendency (mean, median, and mode) values of each column:

🧠 Academic Performance (Percentages)
Metric	SSC (%)	HSC (%)	Degree (%)	E-Test (%)	MBA (%)
Mean	67.30	66.33	66.37	72.10	62.28
Median	67.00	65.00	66.00	71.00	62.00
Mode	62.00	63.00	65.00	60.00	56.70

All mean and median values lie between 60% and 79%, indicating an average academic performance across SSC, HSC, Degree, E-test, and MBA.

E-test performance (mean = 72.10%) stands out as relatively strong among the students.

The mode values reflect the most frequently occurring scores, which are slightly lower, suggesting some students scored on the lower average side in repeated patterns.

💰 Salary Insights
Metric	Salary (INR)
Mean	288,655
Median	265,000
Mode	300,000

The average salary offered to students is around ₹288,655, with the median at ₹265,000, indicating most students received offers in that range.

A mode of ₹300,000 suggests a common offer amount was around 3 Lakhs, which could represent a standard package from one or more recruiters.

✅ **Final Conclusion**

Based on the provided data and your criteria:

* Students scoring **above 60% and below 80%** are considered **average**.
* Since **all academic averages** (mean, median) lie within this range, we conclude that:

> 🟢 **"The overall academic performance of students is average. However, the salary figures suggest a healthy placement trend, indicating employability despite average academic scores."**

In [73]:
 import numpy as np

In [35]:
np.percentile(dataset["ssc_p"],100)

89.4

In [36]:
descriptive=pd.DataFrame(index=["Mean","Median","Mode","Q1:25%","Q2:50%",
                               "Q3:75%","99%","Q4:100%"],columns=quan)
for columnName in quan:
    descriptive[columnName]["Mean"]=dataset[columnName].mean()
    descriptive[columnName]["Median"]=dataset[columnName].median()
    descriptive[columnName]["Mode"]=dataset[columnName].mode()[0]
    descriptive[columnName]["Q1:25%"]=dataset.describe()[columnName]["25%"]
    descriptive[columnName]["Q2:50%"]=dataset.describe()[columnName]["50%"]
    descriptive[columnName]["Q3:75%"]=dataset.describe()[columnName]["75%"]
    descriptive[columnName]["99%"]=np.percentile(dataset[columnName],99)
    descriptive[columnName]["Q4:100%"]=dataset.describe()[columnName]["max"]
    
    
    

In [37]:
descriptive

sl_no    ssc_p    hsc_p degree_p  etest_p    mba_p  salary
Mean        108  67.3034  66.3332  66.3702  72.1006  62.2782  288655
Median      108       67       65       66       71       62  265000
Mode          1       62       63       65       60     56.7  300000
Q1:25%     54.5     60.6     60.9       61       60   57.945  240000
Q2:50%      108       67       65       66       71       62  265000
Q3:75%    161.5     75.7       73       72     83.5   66.255  300000
99%      212.86       87    91.86    83.86       97  76.1142     NaN
Q4:100%     215     89.4     97.7       91       98    77.89  940000

In [38]:
descriptive=pd.DataFrame(index=["Mean","Median","Mode","Q1:25%","Q2:50%",
                               "Q3:75%","99%","Q4:100%","IQR","1.5rule","Lesser","Greater","Min","Max"],columns=quan)
for columnName in quan:
    descriptive[columnName]["Mean"]=dataset[columnName].mean()
    descriptive[columnName]["Median"]=dataset[columnName].median()
    descriptive[columnName]["Mode"]=dataset[columnName].mode()[0]
    descriptive[columnName]["Q1:25%"]=dataset.describe()[columnName]["25%"]
    descriptive[columnName]["Q2:50%"]=dataset.describe()[columnName]["50%"]
    descriptive[columnName]["Q3:75%"]=dataset.describe()[columnName]["75%"]
    descriptive[columnName]["99%"]=np.percentile(dataset[columnName],99)
    descriptive[columnName]["Q4:100%"]=dataset.describe()[columnName]["max"]
    descriptive[columnName]["IQR"]=descriptive[columnName]["Q3:75%"]-descriptive[columnName]["Q1:25%"]
    descriptive[columnName]["1.5rule"]=1.5*descriptive[columnName]["IQR"]
    descriptive[columnName]["Lesser"]=descriptive[columnName]["Q1:25%"]-descriptive[columnName]["1.5rule"]
    descriptive[columnName]["Greater"]=descriptive[columnName]["Q3:75%"]+descriptive[columnName]["1.5rule"]
    descriptive[columnName]["Min"]=dataset[columnName].min()
    descriptive[columnName]["Max"]=dataset[columnName].max()
    

In [39]:
descriptive

sl_no    ssc_p    hsc_p degree_p  etest_p    mba_p  salary
Mean        108  67.3034  66.3332  66.3702  72.1006  62.2782  288655
Median      108       67       65       66       71       62  265000
Mode          1       62       63       65       60     56.7  300000
Q1:25%     54.5     60.6     60.9       61       60   57.945  240000
Q2:50%      108       67       65       66       71       62  265000
Q3:75%    161.5     75.7       73       72     83.5   66.255  300000
99%      212.86       87    91.86    83.86       97  76.1142     NaN
Q4:100%     215     89.4     97.7       91       98    77.89  940000
IQR         107     15.1     12.1       11     23.5     8.31   60000
1.5rule   160.5    22.65    18.15     16.5    35.25   12.465   90000
Lesser     -106    37.95    42.75     44.5    24.75    45.48  150000
Greater     322    98.35    91.15     88.5   118.75    78.72  390000
Min           1    40.89       37       50       50    51.21  200000
Max         215     89.4     97.7       91       98    77.89  940000

In [40]:
lesser=[]
greater=[]

for columnName in quan:
    if(descriptive[columnName]["Min"]<descriptive[columnName]["Lesser"]):
        lesser.append(columnName)
    if(descriptive[columnName]["Max"]>descriptive[columnName]["Greater"]):
        greater.append(columnName)
        
    
    

In [41]:
lesser

['hsc_p']

In [42]:
greater

['hsc_p', 'degree_p', 'salary']

In [43]:
for column in lesser:
    dataset[column][dataset[column]<descriptive[column]["Lesser"]]=descriptive[column]["Lesser"]
for column in greater:
    dataset[column][dataset[column]>descriptive[column]["Greater"]]=descriptive[column]["Greater"]


C:\Anaconda3\envs\aiml\lib\site-packages\ipykernel_launcher.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  
C:\Anaconda3\envs\aiml\lib\site-packages\ipykernel_launcher.py:4: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  after removing the cwd from sys.path.


In [44]:
descriptive=pd.DataFrame(index=["Mean","Median","Mode","Q1:25%","Q2:50%",
                               "Q3:75%","99%","Q4:100%","IQR","1.5rule","Lesser","Greater","Min","Max"],columns=quan)
for columnName in quan:
    descriptive[columnName]["Mean"]=dataset[columnName].mean()
    descriptive[columnName]["Median"]=dataset[columnName].median()
    descriptive[columnName]["Mode"]=dataset[columnName].mode()[0]
    descriptive[columnName]["Q1:25%"]=dataset.describe()[columnName]["25%"]
    descriptive[columnName]["Q2:50%"]=dataset.describe()[columnName]["50%"]
    descriptive[columnName]["Q3:75%"]=dataset.describe()[columnName]["75%"]
    descriptive[columnName]["99%"]=np.percentile(dataset[columnName],99)
    descriptive[columnName]["Q4:100%"]=dataset.describe()[columnName]["max"]
    descriptive[columnName]["IQR"]=descriptive[columnName]["Q3:75%"]-descriptive[columnName]["Q1:25%"]
    descriptive[columnName]["1.5rule"]=1.5*descriptive[columnName]["IQR"]
    descriptive[columnName]["Lesser"]=descriptive[columnName]["Q1:25%"]-descriptive[columnName]["1.5rule"]
    descriptive[columnName]["Greater"]=descriptive[columnName]["Q3:75%"]+descriptive[columnName]["1.5rule"]
    descriptive[columnName]["Min"]=dataset[columnName].min()
    descriptive[columnName]["Max"]=dataset[columnName].max()
    

In [45]:
descriptive

sl_no    ssc_p    hsc_p degree_p  etest_p    mba_p  salary
Mean        108  67.3034  66.3347  66.3586  72.1006  62.2782  277649
Median      108       67       65       66       71       62  265000
Mode          1       62       63       65       60     56.7  300000
Q1:25%     54.5     60.6     60.9       61       60   57.945  240000
Q2:50%      108       67       65       66       71       62  265000
Q3:75%    161.5     75.7       73       72     83.5   66.255  300000
99%      212.86       87   91.129    83.86       97  76.1142     NaN
Q4:100%     215     89.4    91.15     88.5       98    77.89  390000
IQR         107     15.1     12.1       11     23.5     8.31   60000
1.5rule   160.5    22.65    18.15     16.5    35.25   12.465   90000
Lesser     -106    37.95    42.75     44.5    24.75    45.48  150000
Greater     322    98.35    91.15     88.5   118.75    78.72  390000
Min           1    40.89    42.75       50       50    51.21  200000
Max         215     89.4    91.15     88.5       98    77.89  390000

In [46]:
lesser=[]
greater=[]

for columnName in quan:
    if(descriptive[columnName]["Lesser"]>descriptive[columnName]["Min"]):
        lesser.append(columnName)
    if(descriptive[columnName]["Greater"]<descriptive[columnName]["Q4:100%"]):
        greater.append(columnName)
        
    
    

In [47]:
lesser

[]

In [48]:
greater

[]

In [49]:
dataset["ssc_p"].value_counts()

62.00    11
63.00    10
67.00     9
52.00     9
73.00     9
         ..
80.60     1
65.20     1
80.40     1
59.96     1
78.50     1
Name: ssc_p, Length: 103, dtype: int64

In [50]:
freqTable=pd.DataFrame(columns=["Unique_Values","Frequency","Relative Frequency","Cusum"])


In [51]:
freqTable

Empty DataFrame
Columns: [Unique_Values, Frequency, Relative Frequency, Cusum]
Index: []

In [52]:
freqTable["Unique_Values"]=dataset["ssc_p"].value_counts().index
freqTable["Frequency"]=dataset["ssc_p"].value_counts().values
freqTable["Relative Frequency"]=(freqTable["Frequency"]/103)
freqTable["Cusum"]=freqTable["Relative Frequency"].cumsum()

In [53]:
def freqTable(colunName,dataset):
    freqTable=pd.DataFrame(columns=["Unique_Values","Frequency","Relative Frequency","Cusum"])
    freqTable["Unique_Values"]=dataset[columnName].value_counts().index
    freqTable["Frequency"]=dataset[columnName].value_counts().values
    freqTable["Relative Frequency"]=(freqTable["Frequency"]/103)
    freqTable["Cusum"]=freqTable["Relative Frequency"].cumsum()
    return freqTable


In [54]:
freqTable("",dataset)

Unique_Values  Frequency  Relative Frequency     Cusum
0        300000.0         22            0.213592  0.213592
1        250000.0         18            0.174757  0.388350
2        390000.0         15            0.145631  0.533981
3        240000.0         15            0.145631  0.679612
4        260000.0          7            0.067961  0.747573
5        200000.0          6            0.058252  0.805825
6        360000.0          6            0.058252  0.864078
7        265000.0          6            0.058252  0.922330
8        220000.0          5            0.048544  0.970874
9        275000.0          5            0.048544  1.019417
10       210000.0          4            0.038835  1.058252
11       270000.0          4            0.038835  1.097087
12       216000.0          3            0.029126  1.126214
13       350000.0          3            0.029126  1.155340
14       230000.0          2            0.019417  1.174757
15       218000.0          2            0.019417  1.194175
16       252000.0          2            0.019417  1.213592
17       236000.0          2            0.019417  1.233010
18       204000.0          2            0.019417  1.252427
19       280000.0          2            0.019417  1.271845
20       276000.0          2            0.019417  1.291262
21       290000.0          1            0.009709  1.300971
22       336000.0          1            0.009709  1.310680
23       264000.0          1            0.009709  1.320388
24       287000.0          1            0.009709  1.330097
25       380000.0          1            0.009709  1.339806
26       295000.0          1            0.009709  1.349515
27       340000.0          1            0.009709  1.359223
28       233000.0          1            0.009709  1.368932
29       225000.0          1            0.009709  1.378641
30       255000.0          1            0.009709  1.388350
31       268000.0          1            0.009709  1.398058
32       320000.0          1            0.009709  1.407767
33       231000.0          1            0.009709  1.417476
34       285000.0          1            0.009709  1.427184
35       278000.0          1            0.009709  1.436893

In [55]:
def Univariate(dataset,quan):
    descriptive=pd.DataFrame(index=["Mean","Median","Mode","Q1:25%","Q2:50%",
                               "Q3:75%","99%","Q4:100%","IQR","1.5rule","Lesser","Greater","Min","Max"],columns=quan)
    for columnName in quan:
        descriptive[columnName]["Mean"]=dataset[columnName].mean()
        descriptive[columnName]["Median"]=dataset[columnName].median()
        descriptive[columnName]["Mode"]=dataset[columnName].mode()[0]
        descriptive[columnName]["Q1:25%"]=dataset.describe()[columnName]["25%"]
        descriptive[columnName]["Q2:50%"]=dataset.describe()[columnName]["50%"]
        descriptive[columnName]["Q3:75%"]=dataset.describe()[columnName]["75%"]
        descriptive[columnName]["99%"]=np.percentile(dataset[columnName],99)
        descriptive[columnName]["Q4:100%"]=dataset.describe()[columnName]["max"]
        descriptive[columnName]["IQR"]=descriptive[columnName]["Q3:75%"]-descriptive[columnName]["Q1:25%"]
        descriptive[columnName]["1.5rule"]=1.5*descriptive[columnName]["IQR"]
        descriptive[columnName]["Lesser"]=descriptive[columnName]["Q1:25%"]-descriptive[columnName]["1.5rule"]
        descriptive[columnName]["Greater"]=descriptive[columnName]["Q3:75%"]+descriptive[columnName]["1.5rule"]
        descriptive[columnName]["Min"]=dataset[columnName].min()
        descriptive[columnName]["Max"]=dataset[columnName].max()
    return descriptive

In [56]:
Univariate(dataset,quan)

sl_no    ssc_p    hsc_p degree_p  etest_p    mba_p  salary
Mean        108  67.3034  66.3347  66.3586  72.1006  62.2782  277649
Median      108       67       65       66       71       62  265000
Mode          1       62       63       65       60     56.7  300000
Q1:25%     54.5     60.6     60.9       61       60   57.945  240000
Q2:50%      108       67       65       66       71       62  265000
Q3:75%    161.5     75.7       73       72     83.5   66.255  300000
99%      212.86       87   91.129    83.86       97  76.1142     NaN
Q4:100%     215     89.4    91.15     88.5       98    77.89  390000
IQR         107     15.1     12.1       11     23.5     8.31   60000
1.5rule   160.5    22.65    18.15     16.5    35.25   12.465   90000
Lesser     -106    37.95    42.75     44.5    24.75    45.48  150000
Greater     322    98.35    91.15     88.5   118.75    78.72  390000
Min           1    40.89    42.75       50       50    51.21  200000
Max         215     89.4    91.15     88.5       98    77.89  390000

In [57]:
dir(dataset["ssc_p"])

['T',
 '_AXIS_ALIASES',
 '_AXIS_IALIASES',
 '_AXIS_LEN',
 '_AXIS_NAMES',
 '_AXIS_NUMBERS',
 '_AXIS_ORDERS',
 '_AXIS_REVERSED',
 '_HANDLED_TYPES',
 '__abs__',
 '__add__',
 '__and__',
 '__array__',
 '__array_priority__',
 '__array_ufunc__',
 '__array_wrap__',
 '__bool__',
 '__class__',
 '__contains__',
 '__copy__',
 '__deepcopy__',
 '__delattr__',
 '__delitem__',
 '__dict__',
 '__dir__',
 '__div__',
 '__divmod__',
 '__doc__',
 '__eq__',
 '__finalize__',
 '__float__',
 '__floordiv__',
 '__format__',
 '__ge__',
 '__getattr__',
 '__getattribute__',
 '__getitem__',
 '__getstate__',
 '__gt__',
 '__hash__',
 '__iadd__',
 '__iand__',
 '__ifloordiv__',
 '__imod__',
 '__imul__',
 '__init__',
 '__init_subclass__',
 '__int__',
 '__invert__',
 '__ior__',
 '__ipow__',
 '__isub__',
 '__iter__',
 '__itruediv__',
 '__ixor__',
 '__le__',
 '__len__',
 '__long__',
 '__lt__',
 '__matmul__',
 '__mod__',
 '__module__',
 '__mul__',
 '__ne__',
 '__neg__',
 '__new__',
 '__nonzero__',
 '__or__',
 '__pos__',
 '__p

In [58]:
dataset["ssc_p"].skew()

-0.13264949031456927

In [59]:
dataset["ssc_p"].kurtosis()

-0.6075102066986688

In [60]:
def Univariate(dataset,quan):
    descriptive=pd.DataFrame(index=["Mean","Median","Mode","Q1:25%","Q2:50%",
                               "Q3:75%","99%","Q4:100%","IQR","1.5rule","Lesser","Greater",
                                    "Min","Max","kurtosis","skew"],columns=quan)
    for columnName in quan:
        descriptive[columnName]["Mean"]=dataset[columnName].mean()
        descriptive[columnName]["Median"]=dataset[columnName].median()
        descriptive[columnName]["Mode"]=dataset[columnName].mode()[0]
        descriptive[columnName]["Q1:25%"]=dataset.describe()[columnName]["25%"]
        descriptive[columnName]["Q2:50%"]=dataset.describe()[columnName]["50%"]
        descriptive[columnName]["Q3:75%"]=dataset.describe()[columnName]["75%"]
        descriptive[columnName]["99%"]=np.percentile(dataset[columnName],99)
        descriptive[columnName]["Q4:100%"]=dataset.describe()[columnName]["max"]
        descriptive[columnName]["IQR"]=descriptive[columnName]["Q3:75%"]-descriptive[columnName]["Q1:25%"]
        descriptive[columnName]["1.5rule"]=1.5*descriptive[columnName]["IQR"]
        descriptive[columnName]["Lesser"]=descriptive[columnName]["Q1:25%"]-descriptive[columnName]["1.5rule"]
        descriptive[columnName]["Greater"]=descriptive[columnName]["Q3:75%"]+descriptive[columnName]["1.5rule"]
        descriptive[columnName]["Min"]=dataset[columnName].min()
        descriptive[columnName]["Max"]=dataset[columnName].max()
        descriptive[columnName]["kurtosis"]=dataset[columnName].kurtosis()
        descriptive[columnName]["skew"]=dataset[columnName].skew()
    return descriptive

In [61]:
Univariate(dataset,quan)

sl_no     ssc_p      hsc_p   degree_p   etest_p     mba_p    salary
Mean         108   67.3034    66.3347    66.3586   72.1006   62.2782    277649
Median       108        67         65         66        71        62    265000
Mode           1        62         63         65        60      56.7    300000
Q1:25%      54.5      60.6       60.9         61        60    57.945    240000
Q2:50%       108        67         65         66        71        62    265000
Q3:75%     161.5      75.7         73         72      83.5    66.255    300000
99%       212.86        87     91.129      83.86        97   76.1142       NaN
Q4:100%      215      89.4      91.15       88.5        98     77.89    390000
IQR          107      15.1       12.1         11      23.5      8.31     60000
1.5rule    160.5     22.65      18.15       16.5     35.25    12.465     90000
Lesser      -106     37.95      42.75       44.5     24.75     45.48    150000
Greater      322     98.35      91.15       88.5    118.75     78.72    390000
Min            1     40.89      42.75         50        50     51.21    200000
Max          215      89.4      91.15       88.5        98     77.89    390000
kurtosis    -1.2  -0.60751  0.0869008 -0.0974897  -1.08858 -0.470723 -0.239837
skew           0 -0.132649   0.162611   0.204164  0.282308  0.313576    0.8067

In [62]:
dataset["ssc_p"].var()

117.2283767354923

In [63]:
dataset["ssc_p"].std()

10.827205398231452

In [64]:
def Univariate(dataset,quan):
    descriptive=pd.DataFrame(index=["Mean","Median","Mode","Q1:25%","Q2:50%",
                               "Q3:75%","99%","Q4:100%","IQR","1.5rule","Lesser","Greater",
                                    "Min","Max","kurtosis","skew","Var","Std"],columns=quan)
    for columnName in quan:
        descriptive[columnName]["Mean"]=dataset[columnName].mean()
        descriptive[columnName]["Median"]=dataset[columnName].median()
        descriptive[columnName]["Mode"]=dataset[columnName].mode()[0]
        descriptive[columnName]["Q1:25%"]=dataset.describe()[columnName]["25%"]
        descriptive[columnName]["Q2:50%"]=dataset.describe()[columnName]["50%"]
        descriptive[columnName]["Q3:75%"]=dataset.describe()[columnName]["75%"]
        descriptive[columnName]["99%"]=np.percentile(dataset[columnName],99)
        descriptive[columnName]["Q4:100%"]=dataset.describe()[columnName]["max"]
        descriptive[columnName]["IQR"]=descriptive[columnName]["Q3:75%"]-descriptive[columnName]["Q1:25%"]
        descriptive[columnName]["1.5rule"]=1.5*descriptive[columnName]["IQR"]
        descriptive[columnName]["Lesser"]=descriptive[columnName]["Q1:25%"]-descriptive[columnName]["1.5rule"]
        descriptive[columnName]["Greater"]=descriptive[columnName]["Q3:75%"]+descriptive[columnName]["1.5rule"]
        descriptive[columnName]["Min"]=dataset[columnName].min()
        descriptive[columnName]["Max"]=dataset[columnName].max()
        descriptive[columnName]["kurtosis"]=dataset[columnName].kurtosis()
        descriptive[columnName]["skew"]=dataset[columnName].skew()
        descriptive[columnName]["Var"]=dataset[columnName].var()
        descriptive[columnName]["Std"]=dataset[columnName].std()
    return descriptive

In [65]:
Univariate(dataset,quan)

sl_no     ssc_p      hsc_p   degree_p   etest_p     mba_p  \
Mean          108   67.3034    66.3347    66.3586   72.1006   62.2782   
Median        108        67         65         66        71        62   
Mode            1        62         63         65        60      56.7   
Q1:25%       54.5      60.6       60.9         61        60    57.945   
Q2:50%        108        67         65         66        71        62   
Q3:75%      161.5      75.7         73         72      83.5    66.255   
99%        212.86        87     91.129      83.86        97   76.1142   
Q4:100%       215      89.4      91.15       88.5        98     77.89   
IQR           107      15.1       12.1         11      23.5      8.31   
1.5rule     160.5     22.65      18.15       16.5     35.25    12.465   
Lesser       -106     37.95      42.75       44.5     24.75     45.48   
Greater       322     98.35      91.15       88.5    118.75     78.72   
Min             1     40.89      42.75         50        50     51.21   
Max           215      89.4      91.15       88.5        98     77.89   
kurtosis     -1.2  -0.60751  0.0869008 -0.0974897  -1.08858 -0.470723   
skew            0 -0.132649   0.162611   0.204164  0.282308  0.313576   
Var          3870   117.228    112.064    53.6047   176.251   34.0284   
Std       62.2093   10.8272     10.586    7.32152    13.276   5.83338   

              salary  
Mean          277649  
Median        265000  
Mode          300000  
Q1:25%        240000  
Q2:50%        265000  
Q3:75%        300000  
99%              NaN  
Q4:100%       390000  
IQR            60000  
1.5rule        90000  
Lesser        150000  
Greater       390000  
Min           200000  
Max           390000  
kurtosis   -0.239837  
skew          0.8067  
Var       2.9446e+09  
Std          54264.1